draft-handrews-jri.md

```diff
@@ -0,0 +1,536 @@
+
+A JRI cache implementation SHOULD:
+
+* Keep track of each resource's media type and handle fragments accordingly, unless the cache is implementing a context specification that guarantees consistent media types
+* Respect any cache controls on retrieved resources to avoid burdening servers with superfluous requests; this can often be deferred to a caching network protocol client
+
+A JRI cache implementation MAY:
+
+* Offer a plugin interface for resolving cache misses, keyed off of the IRI scheme or other IRI components
+    * Such a system SHOULD allow plugins that locate resources other than by retrieving the IRI; for example in a development environment it might be preferable to resolve `https:` resources from the local file system
+    * Registering a plugin is sufficient to consider the JRI cache implementation "non-default" with respect to the requirement to treat IRIs as non-locators by default
+
+### Document set transformations
+
+A common family of use cases involves reading one or more JRI-using documents and transforming them in a way that makes them usable by a wider range of tooling.
+
+#### Reference removal
+
+This use case attempts to remove all references with object-level semantics by replacing the reference objects with their targets, resulting in a plain JSON (or other underlying format) document that requires no knowledge of JRI.
+
+Object-level replacement MAY be implemented by producing a copy of the source document with the reference object replaced by the contents of its target.
+
+Removing references from a set of documents with differing base IRIs, or involving fragments that are to be resolved within a specific primary resource carries additional restrictions.  References MUST be traversed in a depth-first fashion and removed from the leaf to the root, so that each removal takes place in the proper primary resource context and with the proper base IRI.
+
+#### Bundling to JSON Pointer fragment references only
+
+Despite implementing specifications that allow multi-document sets, in practice a significant number of tools only support same-resource references using JSON Pointer fragments (the only fragment syntax listed by name in older JSON Reference specifications).
+
+This form of bundling creates a single document and adjusts all references to only point to the newly bundled locations using JSON Pointers.
+
+#### Bundling with stable references
+
+A set of documents that each set `"$id"` in their root objects, and that only refer to each other using the primary resource identifiers set that way (plus any appropriate fragment syntax) can be bundled into a single compound document that satisfies all references internally without the need to edit them.
```

Comment:
"set that way"  confuses me. what way? Do you mean primary resource identifiers that correspond to the `"$id"` values?